### DISCUSSION TITLE
Controlling Custom Model Selection from VS Code

### DISCUSSION THREAD

**Discussion by taywrobel** [github-early-access/copilot-custom-models-beta#3] [General]
This guide will help you verify that your custom model is being correctly selected and used in VS Code, and provide guidance for reverting back to the base model on a per-user basis, rather than deleting your custom model for everyone.

### Verifying Custom Model Selection

VS Code exposes information about the models available to you and the custom model the IDE has selected in the Copilot Logs. From the command palette (`F1` or `ctrl`+`shift`+`p` on windows, `cmd`+`shift`+`p` on macos) you can find the `GitHub Copilot: Open Logs` command:

![Screenshot 2024-09-11 at 12 24 34 PM](https://github.com/user-attachments/assets/9de6acd2-0ca7-496c-ae08-252499cb30f7)

This will open a log view at the bottom of the window, which will show the models available to your GitHub account:

```
2024-09-11 12:25:54.534 [info] [customModels] Available model: copilot-prod-finetune-centralus.github-662oo1401450
2024-09-11 12:25:54.534 [info] [customModels] Available model: copilot-codex
2024-09-11 12:25:54.540 [info] [auth] Sucessfully authenticated
```

Custom models have the format `copilot-prod-finetune-centralus.{organization}-{hash}`, so in this case a custom model for the `github` organization is available, as well as the base model, titled `copilot-codex`. You can verify that the correct model is being used by checking the logs after performing a completion:

```
2024-09-11 12:28:35.583 [info] [fetchCompletions] request.response: [https://proxy.enterprise.githubcopilot.com/v1/engines/copilot-prod-finetune-centralus.github-662oo1401450/completions] took 387 ms
2024-09-11 12:28:35.587 [info] [streamChoices] solution 0 returned. finish reason: [stop]
```

### Overriding Model Selection

If you would like to override the model selection to explicitly use the base model rather than your custom model, this can be done in your user or workspace settings JSON. Again open the command palette, and search for `Preferences: Open Workspace Settings (JSON)`.

This will open the settings.json file for your IDE workspace, and you can change the model you're using by setting the value:
```json
{
    "github.copilot.advanced": {
        "debug.overrideEngine": "copilot-codex"
    }
}
```

**Comment by wbond** [comment]
Is this available for JetBrains products also?
